pkg/services/ngalert/notifier/crypto.go

```diff
@@ -36,6 +38,79 @@ func NewCrypto(secrets secrets.Service, configs configurationStore, log log.Logg
 	}
 }
 
+// ProcessReceivers encrypts new secure settings, loads existing secure settings from the database, and assigns missing uids.
+func (c *alertmanagerCrypto) ProcessReceivers(ctx context.Context, orgId int64, recvs []*definitions.PostableApiReceiver) error {
+	// First, we encrypt the new or updated secure settings. Then, we load the existing secure settings from the database
+	// and add back any that weren't updated.
+	// We perform these steps in this order to ensure the hash of the secure settings remains stable when no secure
+	// settings were modified.
+	if err := EncryptReceiverConfigs(recvs, func(ctx context.Context, payload []byte) ([]byte, error) {
+		return c.Encrypt(ctx, payload, secrets.WithoutScope())
+	}); err != nil {
+		return fmt.Errorf("failed to encrypt receivers: %w", err)
+	}
+
+	if err := c.LoadSecureSettings(ctx, orgId, recvs); err != nil {
+		return err
+	}
+
+	if err := assignReceiverConfigsUIDs(recvs); err != nil {
```

Comment:
I am not sure that this is crypto's responsibility.

Reply:
That's fair, I can move it out of `ProcessReceivers` then give it a better name like `ProcessSecureSettings`. I doubt this is even necessary for testing receivers... if not, I can keep it as a function in `alertmanager_config`.

Reply:
c29df0960482ef4f2e4cf65701ad00c6d43cb97b